hpy/devel/__init__.py

```diff
@@ -53,6 +53,15 @@ def fix_distribution(self, dist):
         base_build_ext = dist.cmdclass.get("build_ext", build_ext)
         orig_bdist_egg_write_stub = bdist_egg_mod.write_stub
 
+        assert ('setuptools.command.build_ext', 'build_ext') in [
+            (c.__module__, c.__name__) for c in base_build_ext.__mro__
+        ], (
+            "dist.cmdclass['build_ext'] does not inherit from"
+            " setuptools.command.build_ext.build_ext. The HPy build"
+            " system does not currently support any other build_ext"
+            " classes."
+        )
+
```

Comment:
sorry for the late review.
This is ugly but I agree it's the best we can do. 
We could improve the error message, though. Something like `if you are using distutils.commands.build_ext, please use setuptools.commands.build_ext`

Reply:
Good idea. I push the improved error message straight to master in @06bde571321218d8f6466f05257b680ba1d65fd1.